src/components/order-service/order-service.controller.ts

```diff
@@ -72,7 +93,34 @@ export class OrderServiceController implements angular.IController {
     return;
   }
 
-  public cancelOrder() {
-    return;
+  public closePanel() {
+    if (angular.isFunction(this.ctrl.handleClose)) {
+      this.ctrl.handleClose();
+    }
+  }
+
+  private listProjects() {
+    this.DataService.list('projects', this.$scope).then((response: any) => {
+      this.ctrl.projects = _.sortBy(response.by('metadata.name'), this.$filter('displayName'));
+      this.ctrl.selectedProject = _.first(this.ctrl.projects);
+    });
+  }
+
+  private makeServiceInstance() {
+    let serviceClassName = _.get(this, 'ctrl.serviceClass.resource.metadata.name');
+    let serviceInstance = {
+      kind: 'Instance',
+      apiVersion: 'servicecatalog.k8s.io/v1alpha1',
+      metadata: {
+        namespace: this.ctrl.selectedProject.metadata.name,
+        generateName: serviceClassName + '-'
```

Comment:
Pretty sure a trailing `-` is used with generate name frequently. E.g.

```
apiVersion: v1
kind: Pod
metadata:
  name: hello-openshift-1-mdd3f
  generateName: hello-openshift-1-
```

>  may be truncated by the **length of the suffix**

"Length of the suffix" implies to me it's truncated to meet length limits, but I could be wrong.